### PR TITLE
New Resources: r/`aws_networkmanager_vpc_attachment` & r/`aws_networkmanager_attachment_acceptor`

### LINKED ISSUE #26226: r/`aws_networkmanager_vpc_attachment`
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* aws_networkmanager_vpc_attachment

### Potential Terraform Configuration

```hcl
resource "aws_networkmanager_vpc_attachment" "example" {
  subnet_arns             = [aws_subnet.example.arn]
  core_network_id         = awscc_networkmanager_core_network.example.id
  vpc_arn                 = aws_vpc.example.arn

  options {
    ipv6_support =  false
  }

  tags = {
    "Name" : "my-vpc",
    "segment": "example"
  }
}
```

### References

* https://github.com/hashicorp/terraform-provider-awscc/issues/586

### PR DESCRIPTION
**Pull request by drewmullen**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

 Closes #26226

Output from acceptance testing:

```
$ make testacc TESTS=TestAccNetworkManagerVpcAttachment PKG=networkmanager
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 20 -run='TestAccNetworkManagerVpcAttachment'  -timeout 180m
=== RUN   TestAccNetworkManagerVpcAttachment_basic
=== PAUSE TestAccNetworkManagerVpcAttachment_basic
=== RUN   TestAccNetworkManagerVpcAttachment_tags
=== PAUSE TestAccNetworkManagerVpcAttachment_tags
=== CONT  TestAccNetworkManagerVpcAttachment_basic
=== CONT  TestAccNetworkManagerVpcAttachment_tags
--- PASS: TestAccNetworkManagerVpcAttachment_tags (1394.03s)
--- PASS: TestAccNetworkManagerVpcAttachment_basic (1623.94s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager     1627.529s
...
```

## Notes

- Accepting an attachment is required in order to delete the attachment. Attachments may not be deleted while they have a pending state `AttachmentStatePendingTagAcceptance` or `AttachmentStatePendingAttachmentAcceptance`
- Tests for vpc-attachment have be put into the testfile for `aws_networkmanager_attachment_acceptor` because of the above requirement and because they are effectively the same tests required
- There is currently a bug regarding updating the argument `ipv6_support`. It is not possible to currently update alone, must be updated with subnet_arn

### CODE REVIEW

**Comment by drewmullen** [comment]
```shell
$ make testacc TESTS=TestAccNetworkManagerAttachmentAcceptor PKG=networkmanager
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 20 -run='TestAccNetworkManagerAttachmentAcceptor'  -timeout 180m
=== RUN   TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentBasic
=== PAUSE TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentBasic
=== RUN   TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentTags
=== PAUSE TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentTags
=== CONT  TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentBasic
=== CONT  TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentTags
--- PASS: TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentTags (1247.46s)
--- PASS: TestAccNetworkManagerAttachmentAcceptor_vpcAttachmentBasic (1403.50s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager     1406.898s
```

File: .changelog/26227.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+```release-note:new-resource
+aws_networkmanager_vpc_attachment
+```
+
+```release-note:new-resource
+aws_networkmanager_attachment_acceptor
```
Comment (ewbankkit) [inline .changelog/26227.txt:6]: Should be named `aws_networkmanager_attachment_accepter` 😄.

Comment (ewbankkit) [reply .changelog/26227.txt:6]: https://english.stackexchange.com/questions/317122/difference-between-accepter-and-acceptor

Other similar resource use _er_:

```console
% grep accept internal/provider/provider.go 
			"aws_detective_invitation_accepter": detective.ResourceInvitationAccepter(),
			"aws_dx_hosted_private_virtual_interface_accepter": directconnect.ResourceHostedPrivateVirtualInterfaceAccepter(),
			"aws_dx_hosted_public_virtual_interface_accepter":  directconnect.ResourceHostedPublicVirtualInterfaceAccepter(),
			"aws_dx_hosted_transit_virtual_interface_accepter": directconnect.ResourceHostedTransitVirtualInterfaceAccepter(),
			"aws_directory_service_shared_directory_accepter": ds.ResourceSharedDirectoryAccepter(),
			"aws_ec2_transit_gateway_peering_attachment_accepter":  ec2.ResourceTransitGatewayPeeringAttachmentAccepter(),
			"aws_ec2_transit_gateway_vpc_attachment_accepter":      ec2.ResourceTransitGatewayVPCAttachmentAccepter(),
			"aws_vpc_endpoint_connection_accepter":                 ec2.ResourceVPCEndpointConnectionAccepter(),
			"aws_vpc_peering_connection_accepter":                  ec2.ResourceVPCPeeringConnectionAccepter(),
			"aws_guardduty_invite_accepter":            guardduty.ResourceInviteAccepter(),
			"aws_macie2_invitation_accepter":                 macie2.ResourceInvitationAccepter(),
			"aws_ram_resource_share_accepter": ram.ResourceResourceShareAccepter(),
			"aws_securityhub_invite_accepter":            securityhub.ResourceInviteAccepter(),
```

Comment (drewmullen) [reply .changelog/26227.txt:6]: I'm happy to make the change to be in line with the other resources in the provider. For the record, google will correct you based on Oxford's dictionary :)
image

Comment (drewmullen) [reply .changelog/26227.txt:6]: upon further examination, i yield

😭

**Comment by drewmullen** [comment]
```shell
$ make testacc TESTS=TestAccNetworkManagerAttachmentAccepter PKG=networkmanager
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 20 -run='TestAccNetworkManagerAttachmentAccepter'  -timeout 180m
=== RUN   TestAccNetworkManagerAttachmentAccepter_vpcAttachmentBasic
=== PAUSE TestAccNetworkManagerAttachmentAccepter_vpcAttachmentBasic
=== RUN   TestAccNetworkManagerAttachmentAccepter_vpcAttachmentTags
=== PAUSE TestAccNetworkManagerAttachmentAccepter_vpcAttachmentTags
=== CONT  TestAccNetworkManagerAttachmentAccepter_vpcAttachmentBasic
=== CONT  TestAccNetworkManagerAttachmentAccepter_vpcAttachmentTags
--- PASS: TestAccNetworkManagerAttachmentAccepter_vpcAttachmentTags (1301.31s)
--- PASS: TestAccNetworkManagerAttachmentAccepter_vpcAttachmentBasic (1427.20s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager     1431.028s
```

File: internal/service/networkmanager/vpc_attachment_test.go [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+resource "aws_networkmanager_attachment_accepter" "test" {
+  attachment_id   = aws_networkmanager_vpc_attachment.test.id
+  attachment_type = aws_networkmanager_vpc_attachment.test.attachment_type
+}
+`, azs, ipv6Support)
+}
+
+func testAccCoreNetworkConfig_oneTag(tagKey1, tagValue1 string) string {
+	return TestAccVPCConfig_multipleSubnets +
```
Comment (AdamTylerLynch) [inline internal/service/networkmanager/vpc_attachment_test.go:289]: Please use acctest.ConfigCompose()

**Review by AdamTylerLynch — CHANGES_REQUESTED** [review]
a few small changes requested

**Comment by drewmullen** [comment]
```shell
$ make testacc TESTS=TestAccNetworkManagerVPCAttachment PKG=networkmanager
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 20 -run='TestAccNetworkManagerVPCAttachment'  -timeout 180m
=== RUN   TestAccNetworkManagerVPCAttachment_basic
=== PAUSE TestAccNetworkManagerVPCAttachment_basic
=== RUN   TestAccNetworkManagerVPCAttachment_updates
=== PAUSE TestAccNetworkManagerVPCAttachment_updates
=== RUN   TestAccNetworkManagerVPCAttachment_tags
=== PAUSE TestAccNetworkManagerVPCAttachment_tags
=== CONT  TestAccNetworkManagerVPCAttachment_basic
=== CONT  TestAccNetworkManagerVPCAttachment_tags
=== CONT  TestAccNetworkManagerVPCAttachment_updates
--- PASS: TestAccNetworkManagerVPCAttachment_basic (1216.44s)
--- PASS: TestAccNetworkManagerVPCAttachment_tags (1400.23s)
--- PASS: TestAccNetworkManagerVPCAttachment_updates (1470.83s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager     1474.600s
```

2nd set of tests are running for Accepter but its almost identical to `TestAccNetworkManagerVPCAttachment_basic`

**Review by AdamTylerLynch — APPROVED** [review]
LGTM 🚀

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccNetworkManagerVPCAttachment_' PKG=networkmanager ACCTEST_PARALLELISM=2 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 2  -run=TestAccNetworkManagerVPCAttachment_ -timeout 180m
=== RUN   TestAccNetworkManagerVPCAttachment_basic
=== PAUSE TestAccNetworkManagerVPCAttachment_basic
=== RUN   TestAccNetworkManagerVPCAttachment_disappears
=== PAUSE TestAccNetworkManagerVPCAttachment_disappears
=== RUN   TestAccNetworkManagerVPCAttachment_tags
=== PAUSE TestAccNetworkManagerVPCAttachment_tags
=== RUN   TestAccNetworkManagerVPCAttachment_update
=== PAUSE TestAccNetworkManagerVPCAttachment_update
=== CONT  TestAccNetworkManagerVPCAttachment_basic
=== CONT  TestAccNetworkManagerVPCAttachment_tags
--- PASS: TestAccNetworkManagerVPCAttachment_basic (849.56s)
=== CONT  TestAccNetworkManagerVPCAttachment_disappears
--- PASS: TestAccNetworkManagerVPCAttachment_tags (1123.64s)
=== CONT  TestAccNetworkManagerVPCAttachment_update
--- PASS: TestAccNetworkManagerVPCAttachment_disappears (854.64s)
--- PASS: TestAccNetworkManagerVPCAttachment_update (1284.73s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager	2412.205s
```

**Comment by ewbankkit** [comment]
@drewmullen Thanks for the contribution :tada: :clap:.
Overall this is great.
I made a couple of minor adjustments:

* Removed the `aws_networkmanager_attachment_accepter` acceptance tests as the resource is being tested as part of `aws_networkmanager_vpc_attachment` tests
* Implemented resource Read for `aws_networkmanager_attachment_accepter`

**Pull request merged**